include/aws/common/trace_event.h

```diff
@@ -0,0 +1,108 @@
+#ifndef AWS_COMMON_TRACE_EVENT_H
+#define AWS_COMMON_TRACE_EVENT_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/bus.h>
+#include <aws/common/cJSON.h>
+#include <aws/common/common.h>
+#include <aws/common/process.h>
+#include <aws/common/thread.h>
+AWS_EXTERN_C_BEGIN
+
+/* Must use INIT before calling any other macros*/
+#define AWS_TRACE_EVENT_INIT(address) aws_trace_event_init(address, aws_default_allocator())
+#define AWS_TRACE_EVENT_INIT_(address, allocator) aws_trace_event_init(address, allocator)
+
+/* For use in cpp programs */
+#define AWS_TRACE_EVENT_BEGIN_CPP(category, name) aws_trace_event_new(strdup(category), strdup(name), EVENT_PHASE_BEGIN)
```

Comment:
Instead of duping this, create one on the stack, even if it's static, and pass it in. Use the do { ... } while(0) style macro to execute a block.